### PR TITLE
Phpstan update

### PR DESCRIPTION
**Pull request by sreichel**
### Description

- several phpstan fixes
- updadet phpstan to 1.9.2
- updated workflow
- added to default checks
  - app/code/core/Mage/Centinel
  - app/code/core/Mage/PaypalUk
  - app/code/core/Mage/Use
  - app/code/core/Mage/Weee
  - lib/Mage 
  - lib/Magento
  - lib/Varien
- some CS-fixes (sorry)
  - lib/Mage
  - lib/Magento
  - lib/Varien

### Note

[Last commit](https://github.com/OpenMage/magento-lts/pull/2693/commits/3d547151beff7928b4081cf9d81eae19f0d0c5e2) changed phpstan level from 3 to 5. [Level 6](https://phpstan.org/user-guide/rule-levels) isn't possible b/c missing typehints.

#### Cons:
- huge baseline file

#### Pro
- all fixable errors listed
- checks should complete faster w/o experimental check

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Product/Image.php [hunk 1]
```diff
@@ -250,14 +250,17 @@ protected function _getMemoryLimit()
...
-            return substr($memoryLimit, 0, -1) * 1024;
+        $unit = substr($memoryLimit, -1);
+        $value = (int)substr($memoryLimit, 0, -1);
+
+        if ($unit == 'K') {
+            return $value * 1024;
         }
-        if (substr($memoryLimit, -1) == 'M') {
-            return substr($memoryLimit, 0, -1) * 1024 * 1024;
+        if ($unit == 'M') {
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Product/Image.php:259]:
```suggestion
        if ($unit === 'M') {
```

File: app/code/core/Mage/Catalog/Model/Product/Image.php [hunk 2]
```diff
@@ -250,14 +250,17 @@ protected function _getMemoryLimit()
...
+            return $value * 1024 * 1024;
         }
-        if (substr($memoryLimit, -1) == 'G') {
-            return substr($memoryLimit, 0, -1) * 1024 * 1024 * 1024;
+        if ($unit == 'G') {
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Product/Image.php:262]:
```suggestion
        if ($unit === 'G') {
```

File: app/code/core/Mage/Eav/Model/Convert/Adapter/Grid.php [hunk 3]
```diff
@@ -38,7 +38,7 @@ class Mage_Eav_Model_Convert_Adapter_Grid extends Mage_Dataflow_Model_Convert_Ad
     protected $_entityType;
 
     /**
-     * @return Mage_Eav_Model_Entity_Interface
+     * @return string
```
Comment (tmotyl) [inline app/code/core/Mage/Eav/Model/Convert/Adapter/Grid.php:41]: This change doesnt sound right

File: app/code/core/Mage/Rule/Model/Condition/Abstract.php [hunk 4]
```diff
@@ -28,7 +28,7 @@
  *
  * @method string|false getAttribute()
  * @method $this setAttribute(string|false $value)
- * @method array getAttributeOption()
+ * @method array|string getAttributeOption()
```
[the remark this answers is not in the document]
Comment (sreichel) [reply app/code/core/Mage/Rule/Model/Condition/Abstract.php]: Tested ...

**Comment by sreichel** [comment]
@tmotyl there are no changes in baseline between 1.9.1 (https://github.com/OpenMage/magento-lts/pull/2693/commits/e459410f516f29b8a844244cfd25defb7f30e863) and 1.9.2. I'll update with next PR.

**Review by kiatng — APPROVED** [review]

**Comment by sreichel** [comment]
@tmotyl any ideas for `Cannot declare class XMLParser` error?

File: app/code/core/Mage/Centinel/Block/Adminhtml/Validation.php [hunk 5]
```diff
@@ -64,11 +64,9 @@ protected function _toHtml()
         if (!$payment->getMethod()
             || !$payment->getMethodInstance()
             || $payment->getMethodInstance()->getIsDummy()
-            || !$payment->getMethodInstance()->getIsCentinelValidationEnabled())
-        {
+            || !$payment->getMethodInstance()->getIsCentinelValidationEnabled()) {
```
Comment (kiatng) [inline app/code/core/Mage/Centinel/Block/Adminhtml/Validation.php:67]: Not sure about this:
```suggestion
            || !$payment->getMethodInstance()->getIsCentinelValidationEnabled()
        ) {
```

File: app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php [hunk 6]
```diff
@@ -40,14 +40,14 @@ class Mage_CurrencySymbol_Model_System_Currencysymbol
     /**
      * Store id
      *
-     * @var string | null
+     * @var string|null
```
Comment (kiatng) [inline app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php:43]:
```suggestion
     * @var string|int|null
```

File: app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php [hunk 7]
```diff
@@ -40,14 +40,14 @@ class Mage_CurrencySymbol_Model_System_Currencysymbol
...
      */
     protected $_storeId;
 
     /**
      * Website id
      *
-     * @var string | null
+     * @var string|null
```
Comment (kiatng) [inline app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php:50]:
```suggestion
     * @var string|int|null
```

File: app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php [hunk 8]
```diff
@@ -80,7 +80,7 @@ class Mage_CurrencySymbol_Model_System_Currencysymbol
     /**
      * Sets store Id
      *
-     * @param  int $storeId
+     * @param  string $storeId
```
Comment (kiatng) [inline app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php:83]:
```suggestion
     * @param  string|int $storeId
```

File: app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php [hunk 9]
```diff
@@ -94,7 +94,7 @@ public function setStoreId($storeId = null)
     /**
      * Sets website Id
      *
-     * @param  int $websiteId
+     * @param  string $websiteId
```
Comment (kiatng) [inline app/code/core/Mage/CurrencySymbol/Model/System/Currencysymbol.php:97]:
```suggestion
     * @param  string|int $websiteId
```

**Comment by sreichel** [comment]
I cant reproduce this error in local environment, so i've to test it via PR ...

- downgrade phpstan to 1.9.1
- exclude libs step by step
- ....

**Comment by sreichel** [comment]
Going to split this for easier review ...

**Pull request closed without merging**